src/frontend/Navigation/AppNavigator.tsx

```diff
@@ -28,13 +33,20 @@ import {useNavigationFromRoot} from '../hooks/useNavigationWithTypes';
 
 export const AppNavigator = () => {
   const {formatMessage} = useIntl();
-  const navigation = useNavigationFromRoot();
+  const isLoaded = useEverythingLoaded();
+  const deviceName = useDeviceName();
+
+  if (isLoaded && !deviceName.isPending) {
+    BootSplash.hide();
+  }
```

Comment:
this will be used in a subsequent pr to determine what should be loaded: either the map, or the device naming screens